Source/WebCore/platform/graphics/Gradient.h

```diff
@@ -115,10 +121,13 @@ class Gradient final : public RenderingResource {
     sk_sp<SkShader> shader(float globalAlpha, const AffineTransform&);
 #endif
 
-private:
-    Gradient(Data&&, ColorInterpolationMethod, GradientSpreadMethod, GradientColorStops&&, std::optional<RenderingResourceIdentifier>);
+    void addObserver(WeakRef<RenderingResourceObserver>&& observer) const
+    {
+        m_observers.add(WTFMove(observer));
+    }
```

Comment:
What is the plan of the `RenderingResourceObservers`? Is the plan to add similar code in all types of resources. I think this code can be put in some base class instead of repeating it.

Reply:
The idea is to not have a base class, because the types themselves are not related. They also might have slightly different refcounting or weakptr behavior. The code itself is trivial, I do not think there is anything gained by having a the simple code in a shared mixin. 
Later on, the idea is perhaps to replace RenderingResourceObserver with more fine-grained observers, notably NativeImageObserver  and perhaps RenderingResourceObserver. This is unclear ATM, since it's not trivially clear to me if the weakptr factories work then with the same receiver.  The NativeImage counting needs are different to others currently, so it may be that fine-grained observer would be useful.
If the fine-grained observer is not needed, the mixing approach (not a shared base class) could be reintroduced.